contentcuration/contentcuration/frontend/administration/pages/Users/__tests__/emailUsersDialog.spec.js

```diff
@@ -46,128 +95,190 @@ function makeBulkWrapper() {
         location: 'Czech Republic',
         keywords: 'test',
       },
-    },
+    });
+    expect(
+      screen.getByText('All active users from Czech Republic matching "test"'),
+    ).toBeInTheDocument();
   });
-}
 
-describe('emailUsersDialog', () => {
-  let wrapper;
-
-  beforeEach(async () => {
-    wrapper = makeWrapper();
-    await wrapper.setProps({ value: true }); // Allow watch event to trigger
+  it('shows placeholder buttons', () => {
+    renderComponent();
+    ['First name', 'Last name', 'Email', 'Date', 'Time'].forEach(text =>
+      expect(screen.getByText(text)).toBeInTheDocument(),
+    );
   });
 
-  it('recipients should get set to userIds on dialog open', () => {
-    expect(wrapper.vm.recipients).toEqual([userId, userId2]);
-  });
+  describe('form validation', () => {
+    it('shows validation errors when submitting empty form', async () => {
+      const user = userEvent.setup();
+      renderComponent({ initialRecipients: [userId] });
 
-  describe('on close', () => {
-    it('should emit input event with false value', () => {
-      wrapper.vm.show = false;
-      expect(wrapper.emitted('input')[0][0]).toBe(false);
-    });
+      await user.click(screen.getByText('Send email'));
```

Comment:
`getByRole` would be better for buttons. It's more specific and aligned with VTL's query priority guidance. I think there are more places like this in the test suite. I will point to some of them but likely not all - please revisit the file.